rotkehlchen/chain/ethereum/interfaces/ammswap/ammswap.py

```diff
@@ -321,7 +325,7 @@ def swaps_to_trades(swaps: List[AMMSwap]) -> List[AMMTrade]:
 
     @staticmethod
     def _update_assets_prices_in_address_balances(
-            address_balances: AddressToLPBalances,
+            address_balances: Union[AddressToLPBalances, 'AddressToUniswapV3LPBalances'],
```

Comment:
Uniswap V3 adds additional fields to the `LiquidityPool` object. As seen in `NFTLiquidityPool`

Reply:
Then I think the approach of using the same base function in ammswap.py baseclass may not be the best.

Reply:
I agree also. Duplicating function is also bad because the it's basically the same code but changing the `address_balances` type.

Reply:
So try to find a way to abstract away common functionality

Reply:
Abstracting common functionality is basically moving the entire function content to a different module.
Because the entire content is used for both `AddressToLPBalances` & `AddressToUniswapV3LPBalances`.

I think abstraction in this context might not be needed.